### ISSUE TITLE
[java] UnusedPrivateMethod false-positive used in lambda

### ISSUE THREAD

**Issue by blacelle** [pmd/pmd#4817]
**Affects PMD Version:** 
7.0.0-rc4

**Rule:** [UnusedPrivateMethod](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**

False-positive as `notCachedGetAllDps` is used

```
[INFO] --- maven-pmd-plugin:3.21.2:check (default-cli) @ web-test ---
[INFO] PMD version: 7.0.0-rc4
[INFO] PMD Failure: io.mitrust.backoffice.NotUsedPrivateMethodFalsePositive:15 Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'notCachedGetAllDps(String)'
```

**Code Sample demonstrating the issue:**

```java
package io.mitrust.backoffice;

import java.util.Collections;
import java.util.List;

import com.google.common.cache.CacheBuilder;
import com.google.common.cache.CacheLoader;
import com.google.common.cache.LoadingCache;

public class NotUsedPrivateMethodFalsePositive {

    private final LoadingCache<String, List<? extends String>> clientIdsToDps = CacheBuilder.newBuilder()
            .build(CacheLoader.from(clientId -> notCachedGetAllDps(clientId)));

    private List<? extends String> notCachedGetAllDps(String clientId) {
        return Collections.singletonList(clientId);
    }

    public List<?> getClientIdsToDps() {
        return clientIdsToDps.getUnchecked("");
    }

}

```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by danjee** [comment]
I've found the issue is non lambda calls as-well.

**Comment by danjee** [comment]
Also this happens in imports. If you use an import like `Enum.valueOf` in a lambda rule="UnnecessaryImport" kicks in.
